pkg/provider/cloud/kubevirt/data_volume.go

```diff
@@ -23,49 +23,121 @@ import (
 	cdiv1beta1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
 
 	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	kuberneteshelper "k8c.io/kubermatic/v2/pkg/kubernetes"
+	"k8c.io/kubermatic/v2/pkg/provider"
 	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
 
 	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	utilpointer "k8s.io/utils/pointer"
 	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	preAllocatedDVAnnotation  = "cdi.kubevirt.io/storage.deleteAfterCompletion"
+	preAllocatedDVSize        = "11Gi"
+	osAnnotationForCustomDisk = "cdi.kubevirt.io/os-type"
+	dataVolumeClusterRole     = "datavolume-cloner"
+	dataVolumeRoleBinding     = "allow-datavolume-cloning"
+)
+
 func dataVolumeCreator(datavolume *cdiv1beta1.DataVolume) reconciling.NamedCDIv1beta1DataVolumeCreatorGetter {
 	return func() (name string, create reconciling.CDIv1beta1DataVolumeCreator) {
 		return datavolume.Name, func(dv *cdiv1beta1.DataVolume) (*cdiv1beta1.DataVolume, error) {
+			dv.Annotations = datavolume.Annotations
 			dv.Spec = datavolume.Spec
 			return dv, nil
 		}
 	}
 }
 
-func reconcilePreAllocatedDataVolumes(ctx context.Context, cluster *kubermaticv1.Cluster, client ctrlruntimeclient.Client) error {
+func reconcileDataVolume(ctx context.Context, client ctrlruntimeclient.Client, dataVolume *cdiv1beta1.DataVolume, namespace string) error {
+	dvCreator := []reconciling.NamedCDIv1beta1DataVolumeCreatorGetter{
+		dataVolumeCreator(dataVolume),
+	}
+	return reconciling.ReconcileCDIv1beta1DataVolumes(ctx, dvCreator, namespace, client)
+}
+
+// reconcileLocalDataVolumes reconciles the local-disks from cluster.
+func reconcileLocalDataVolumes(ctx context.Context, cluster *kubermaticv1.Cluster, client ctrlruntimeclient.Client) error {
 	for _, d := range cluster.Spec.Cloud.Kubevirt.PreAllocatedDataVolumes {
-		dv, err := createPreAllocatedDataVolume(d, cluster.Status.NamespaceName)
+		dv, err := newDataVolume(d, cluster.Status.NamespaceName)
 		if err != nil {
 			return err
 		}
-		dvCreator := []reconciling.NamedCDIv1beta1DataVolumeCreatorGetter{
-			dataVolumeCreator(dv),
+		if err := reconcileDataVolume(ctx, client, dv, cluster.Status.NamespaceName); err != nil {
+			return fmt.Errorf("failed to reconcile DataVolume: %w", err)
+		}
+	}
+	return nil
+}
+
+// reconcilePreAllocatedDataVolume reconciles the preAllocated DataVolumes from standard VM images if cloning is enabled.
+func reconcilePreAllocatedDataVolume(ctx context.Context, dc *kubermaticv1.DatacenterSpecKubevirt, client ctrlruntimeclient.Client) error {
+	if dc.Images == nil || dc.Images.HTTP == nil || dc.Images.HTTP.ImageCloning == nil || !dc.Images.HTTP.ImageCloning.Enable {
+		return nil
```

Comment:
Should we not remove all PreAllocated DV in `kubevirt-disks` ? Or even Custom disks (if Cloning is disabled, de we want to keep the Custom Disks in there) ?

Reply:
If it is disabled we leave them as it is.

Reply:
why do not cleaning ? What happens if we have several seeds using the same Kv infra cluster ?
One of them is `enable`, the other one is `disable`....
Should we have a `kubevirt-disks-<seed-name>` ?

Reply:
List VMImages API response will change based on the DC context.
If cloning is disabled in one kubevirt DC, API will return something like <os-version>: <url>

If cloning is enabled in other kubevirt DC, API will return: <os-version>: <kubevirt-disk/DVName>

Reply:
It would still be possible to shoot a Create Clusters request/Create MD using cloning on a dc where it's disabled.
Reconciliation in the `kubevirt-disks` namespace is done (or before disabling or through another dc where it's enabled).
Image DV are there, Role and RoleBinding too. 
Would work even though it has been disabled.

Reply:
List VM images API will return `osversion : url` in case of cloning is disabled and users will select from these options in UI.

If cloning is enabled API will return `osversion : kubevirt-disk/dvName`, so user will select from these options.

Reply:
let's discuss this during the sig-virt meeting ?

Reply:
okay.